File: backend/src/main/java/com/contractly/clause/model/Clause.java
package com.contractly.clause.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Clause {
    private Long id;
    private Long userId;
    private String title;
    private String category;    // "payment_terms", "ip_ownership", "nda", etc.
    private String content;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: backend/src/models/Clause.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface IClause extends Document {
  userId: mongoose.Types.ObjectId;
  title: string;
  category?: string; // e.g., "payment_terms", "ip_ownership", "nda"
  content: string;
  createdAt: Date;
  updatedAt: Date;
}

const ClauseSchema: Schema = new Schema(
  {
    userId: { type: Schema.Types.ObjectId, ref: 'User', required: true },
    title: { type: String, required: true },
    category: { type: String },
    content: { type: String, required: true },
  },
  { timestamps: true }
);

export default mongoose.model<IClause>('Clause', ClauseSchema);
